save-demo/src/main/kotlin/com/saveourtool/save/demo/entity/Dependency.kt

```diff
@@ -0,0 +1,22 @@
+package com.saveourtool.save.demo.entity
+
+import com.saveourtool.save.spring.entity.BaseEntity
+import javax.persistence.Entity
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToOne
+import javax.persistence.Table
+
+/**
+ * @property coreTool dependant tool
+ * @property dependencyTool the tool that [coreTool] depends on
+ */
+@Entity
+@Table(name = "dependency")
+class Dependency(
```

Comment:
it's a specific diktat cases -- do we really want to support it?
we can store a tool as a single artifact (even in case of diktat): as archive (tar, zip)

Reply:
Do we want to think of some universal format for saveourtool demo uploading for now?

Reply:
I mean, probably we would like it to be an automated system. I am not sure if we would let developer upload an archive that contains everything in it.

Reply:
We can automate uploading artifacts from github. Another system will be required another approach for uploading.

About universal format: I don't like the idea with dependencies. It's our specific case that we use ktlint which is a tool too.
Diktat can be converted to a single executable file (we actually already have a task for it). And as I remember, we wanted to limit our tools to a single executable file (but it's for save-cli, not a demo).

Can be left for now, but I think we will revise it in future

Reply:
The truth is that I do not like the solution also but I could not come up with a better idea

Reply:
Completely removed `Dependency` and everything connected with it. Later will add two files: `setup.sh` and `run.sh` or something like that